fairscale/experimental/nn/data_parallel/gossip/mixing_manager.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Mixing Manager Class
+
+:description: Class provides an API for dynamically selecting mixing weights
+              for gossip
+"""
+
+from typing import Dict, Optional, Union
+
+import torch
+
+from .graph_manager import GraphManager
+
+
+class MixingManager(object):
+    def __init__(self, graph: GraphManager, device: Optional[torch.device]) -> None:
+        self.graph_manager = graph
+        self.device = device
+
+    def is_regular(self) -> bool:
+        """
+        Whether there is bias accumulated in local entry of stationary
+        distribution of mixing matrix
+        """
+        return self.graph_manager.is_regular_graph() and self.is_uniform()
+
+    def is_uniform(self) -> bool:
```

Comment:
nit: you can use [abc.abstract base class](https://docs.python.org/3/library/abc.html) to signal that also, maybe more IDE support and more pythonic (mypy will catch an offending class, vs. here it will be caught at runtime), but not nitpicking